src/classes/UTIL_AbstractCommon_BATCH.cls

```diff
@@ -0,0 +1,521 @@
+     * @param batchSize Integer Size of each batch
+     * @param ldvMode Boolean, if true will execute in LDV Chunking mode by default
+     * @param additionalWhereClause if present, will be applied to the inner query, or the main query if executing in non ldv mode.
+     *                               If a main where clause is also present both will be encapsulated in () and anded.
+     */
+    public UTIL_AbstractCommon_BATCH(Integer batchSize, Boolean ldvMode, String additionalWhereClause) {
+        this.endReached = false;
+        this.batchSize = batchSize;
+        this.chunkIteration = 0;
+        this.ldvMode = ldvMode;
+        this.additionalWhereClause = additionalWhereClause;
+    }
+
+    /**
+     * @description start method for the Database batch.  Overloaded to run in LDV Chunking or regular batch mode depending on parameters supplied, and/or
+     * record distance between minimum and maximum Id for the main object of the batch.  If in LDV/Chunking mode it will be the querylocator for the
+     * current chunk of records
+     * @param bc BatchableContext passed from Salesforce
+     * @return Database.QueryLocator Querylocator for batch
+     */
+    public Database.QueryLocator start(Database.BatchableContext bc) {
+        onStart(bc, isFirstChunk());
+
+        if (startId == null) {
+            this.startId = determineStartId();
+            this.endId = determineEndId();
+            this.ldvMode = (ldvMode != true) ? isLDVCircumstance(this.startId, this.endId) : ldvMode;
+        }
+
+        Database.QueryLocator ql = null;
+        Database.QueryLocatorIterator qli = null;
+
+        // If startId is null, there were no records present in the object
+        if (startId != null) {
+            ql = Database.getQueryLocator(getQuery(startId));
+            qli = (ql != null) ? ql.iterator() : null;
+        }
+
+        // If no records were present/returned, it will execute a limited query with a LIMIT 0 in order to avoid
+        // Iterable errors
+        if ((qli == null) || (!qli.hasNext())) {
+            ql = Database.getQueryLocator(getQueryLimitZero());
+            this.endReached = true;
+        }
+
+        return ql;
+    }
+
+    /**
+     * @description Query to be used if object has no records to prevent not iterable error
+     * @return String queryString to execute for Object, limited to 0 records
+     */
+    @testVisible private String getQueryLimitZero() {
+        return 'SELECT Id FROM ' + getSObjectName() + ' LIMIT 0';
+    }
+
+    /**
+     * @description Determines the query to execute factoring all the different params for where clause, ldv versus non ldv, etc.
+     * @param startId starting id of the chunk
+     * @return String queryString to execute for Batch
+     */
+    @testVisible private String getQuery(Id startId) {
+        return 'SELECT ' + getQueryFieldList() + ' FROM ' + getSObjectName() + getQueryWhereClause() + getQueryOrderByAndLimitClause();
```

Comment:
Rather than have to add a blank space before every occurrence of "WHERE" in the code below, why not just add the space into this query builder so it looks like:
```
'SELECT ' + getQueryFieldList() + ' FROM ' + getSObjectName() + ' ' + getQueryWhereClause() + ' ' + getQueryOrderByAndLimitClause();
```